Rename according to GO standards

Darth-Mera opened the pull request:
Renames functions, vars etc. accroding to Go standards and best practices, and introduces some other style changes.

1. unexported identifiers use camelCase
2. public identifiers use PascalCase
3. outputted errors should be lowercase and not end with a punctuation mark
4. format code using go fmt

Signed-off-by: Veronika Fuxova <vfuxova@redhat.com>

Darth-Mera committed:
Rename according to GO standards

Renames functions, vars etc. accroding to Go standards and best
practices, and introduces some other style changes.

1. unexported identifiers use camelCase
2. public identifiers use PascalCase
3. outputted errors should be lowercase and not end with a punctuation
   mark
4. format code using go fmt

Signed-off-by: Veronika Fuxova <vfuxova@redhat.com>

alessandrocarminati commented on config.go:
```diff
@@ -1,267 +1,268 @@
-func func_help		(conf *configuration,fn []string)		(error){
-	return errors.New("Command Help")
+func funcHelp(conf *configuration, fn []string) error {
+	return errors.New("command Help")
```
Is there any reason to change the string case?

Darth-Mera replied:
I have used the refactoring tool in my IDE for renaming, and I overlooked that it also changed the messages. So no actual reason behind it. I can change it back tomorrow

Darth-Mera replied:
Sorry for my previous short and not very clear message, it was typed quickly from my phone. To elaborate: 

I have changed variable names using the IDE refactoring tool and it apparently also changed the message strings in some cases, which I overlooked.

My IDE also changed the first capital letters in the error strings to lowercase to follow this rule: https://google.github.io/styleguide/go/decisions.html#error-strings

That's why there are those discrepancies - relying on automation without a keen eye  :) Thanks for catching it in your review! 

I will change the error strings so they are not so mismatched - I think having them all in lowercase is the way to go. Or do you suggest something else @alessandrocarminati ?

Darth-Mera replied:
@alessandrocarminati I have pushed a commit attempting to unify the strings https://github.com/alessandrocarminati/nav/pull/10/commits/690d6e6929b1408cdd23bcb11d67d403223414d7 

Feel free to offer your thoughts

alessandrocarminati replied:
Looks good!

alessandrocarminati commented on config.go:
```diff
@@ -1,267 +1,268 @@
+	}
+	if s < 1 || s > 2 {
+		return errors.New("unsupported Mode")
+	}
```
Same as before, why change also the message string case?
Is it because you want to match the message string with the variable field name?

alessandrocarminati commented on config.go:
```diff
@@ -1,267 +1,268 @@
-		 return  Default_config, errors.New("Missing switch arg")
-		}
+		return defaultConfig, errors.New("missing switch arg")
+	}
```
Here you change the case of the first letter of the first word. I assume this to be for an esthetic reason. So why you didn't change the case at line 189, and at line123 you made the opposite?

alessandrocarminati replied:
I understand that proper case for identifiers in Golang is crucial for visibility and that there are best practices to follow. I know I have deliberately violated them, like it was in Fortran77, we must be aware that identifier names are more than just names. However, I don't understand the capitalization changes in comments and printed strings. Is there something important that I'm missing?

alessandrocarminati commented on nav.go:
```diff
@@ -1,169 +1,168 @@
+	jsonOutputB64   int    = 3
+	jsonOutputGZB64 int    = 4
+	jsonOutputFMT   string = "{\"graph\": \"%s\",\"graph_type\":\"%s\",\"symbols\": [%s]}"
+)
```
I haven't yet, but my goal is to group these constants together and potentially replace this declaration with something like the example I provided
```
const (
	graphOutputOnly       int    = iota
	jsonOutputPlain 
	jsonOutputB64 
	jsonOutputGZB64 
	outputLast
```
This because these it might need to add view modes and having the last value can be useful.
"I am not sure if Go allows for completely unrelated constants to be declared in the same statement, and even if it does, I feel it might not be appropriate to do so.  
@mauriziopapini do you have any thoughts on this?

alessandrocarminati replied:
I'll merge it as is, and then I need to work on the branching and merge AC-indirect-calls-info.

Darth-Mera committed:
Unify error messages

This PR is a suggestion how the errors and printed messages can be
unified.

I have made these changes:
1. test errors begin with capital letter
2. printed errors are all lowercase
3. I have fixed some capitalization discrepancies in Help

Feel free to comment and suggest what should be done differently.

Signed-off-by: Veronika Fuxova <vfuxova@redhat.com>
